pandas/tests/indexes/common.py

```diff
@@ -635,6 +635,16 @@ def test_delete_base(self):
                 # either depending on numpy version
                 result = idx.delete(len(idx))
 
+    def test_equals(self):
+
+        for name, idx in compat.iteritems(self.indices):
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.astype(object)))
+            self.assertFalse(idx.equals(list(idx)))
```

Comment:
@jorisvandenbossche ``astype`` is introduced by myself, it intends to keep ``DatetimeIndex.equals(object Index with same content)`` result as ``True``. 

But I feel it shouldn't cast at all, because following result is a bit surprising.

```
# on current master
i1 = pd.Index([np.datetime64('2011-01-01')], dtype=object)
i2 = pd.Index([pd.Timestamp('2011-01-01')], dtype=object)
i1.equals(i2)
# False

pd.DatetimeIndex(i1).equals(i2)
# True
```